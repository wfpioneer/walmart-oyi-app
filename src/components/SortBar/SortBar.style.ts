import { StyleSheet } from 'react-native';
import COLOR from '../../themes/Color';

const styles = StyleSheet.create({
  viewSwitcher: {
    backgroundColor: COLOR.WHITE,
    width: '100%',
    alignItems: 'flex-end',
    justifyContent: 'flex-end',
    paddingVertical: 5,
    paddingHorizontal: 15,
    flexDirection: 'row'
  }
});

export default styles;
