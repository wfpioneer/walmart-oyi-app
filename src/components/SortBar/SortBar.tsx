import React from 'react';
import { TouchableOpacity, View } from 'react-native';
import MaterialIcons from 'react-native-vector-icons/MaterialIcons';
import COLOR from '../../themes/Color';
import styles from './SortBar.style';

interface SortBarProps {
  isGrouped: boolean;
  updateGroupToggle: React.Dispatch<React.SetStateAction<boolean>>;
}

const SortBar = ({ isGrouped, updateGroupToggle }: SortBarProps) => (
  <View style={styles.viewSwitcher}>
    <TouchableOpacity onPress={() => updateGroupToggle(false)} testID="menu">
      <MaterialIcons
        name="menu"
        size={25}
        color={!isGrouped ? COLOR.BLACK : COLOR.GREY}
      />
    </TouchableOpacity>
    <TouchableOpacity onPress={() => updateGroupToggle(true)} testID="list">
      <MaterialIcons
        name="list"
        size={25}
        color={isGrouped ? COLOR.BLACK : COLOR.GREY}
      />
    </TouchableOpacity>
  </View>
);

export default SortBar;
