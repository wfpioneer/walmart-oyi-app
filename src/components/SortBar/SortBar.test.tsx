import { fireEvent, render } from '@testing-library/react-native';
import SortBar from './SortBar';

jest.mock('react-native-vector-icons/MaterialIcons', () => 'mockMaterialIcons');

describe('SortBar Component', () => {
  it('Test renders default PalletWorkList Screen', () => {
    const mockUpdateToggle = jest.fn();
    const { toJSON, getByTestId } = render(SortBar(false, mockUpdateToggle));

    const menuToggle = getByTestId('menu');
    const listToggle = getByTestId('list');

    fireEvent.press(menuToggle);
    expect(mockUpdateToggle).toHaveBeenCalledWith(false);

    fireEvent.press(listToggle);
    expect(mockUpdateToggle).toHaveBeenCalledWith(true);
    expect(toJSON()).toMatchSnapshot();
  });
});
