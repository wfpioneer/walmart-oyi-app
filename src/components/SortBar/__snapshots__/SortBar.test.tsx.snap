// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortBar Component Test renders default PalletWorkList Screen 1`] = `
<View
  style={
    {
      "alignItems": "flex-end",
      "backgroundColor": "#ffffffff",
      "flexDirection": "row",
      "justifyContent": "flex-end",
      "paddingHorizontal": 15,
      "paddingVertical": 5,
      "width": "100%",
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
    testID="menu"
  >
    <mockMaterialIcons
      color="#000000"
      name="menu"
      size={25}
    />
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
    testID="list"
  >
    <mockMaterialIcons
      color="#c5c5c5"
      name="list"
      size={25}
    />
  </View>
</View>
`;
